bazel/foreign_cc/luajit.patch

```diff
@@ -65,11 +43,34 @@ index f56465d..3f4f2fa 100644
 
  ##############################################################################
 EOF
-
-patch -p1 < luajit_make.diff
-
-# Default MACOSX_DEPLOYMENT_TARGET is 10.4, which will fail the build at link time on macOS 10.14:
-# ld: library not found for -lgcc_s.10.4
-# This doesn't affect other platforms
-MACOSX_DEPLOYMENT_TARGET=10.6 DEFAULT_CC=${CC} TARGET_CFLAGS=${CFLAGS} TARGET_LDFLAGS=${CFLAGS} \
-  CFLAGS="" make V=1 PREFIX="$THIRDPARTY_BUILD" install
+diff --git a/build.py b/build.py
+new file mode 100755
+index 0000000..9c71271
+--- /dev/null
++++ b/build.py
+@@ -0,0 +1,25 @@
++#!/usr/bin/env python
++
++import argparse
++import os
++
++def main():
++    parser = argparse.ArgumentParser()
++    parser.add_argument("--prefix")
++    args = parser.parse_args()
++    os.chdir(os.path.dirname(os.path.realpath(__file__)))
++
++    os.environ["MACOSX_DEPLOYMENT_TARGET"] = "10.6"
++    os.environ["DEFAULT_CC"] = os.environ.get("CC", "")
++    os.environ["TARGET_CFLAGS"] = os.environ.get("CFLAGS", "")
++    os.environ["TARGET_LDFLAGS"] = os.environ.get("CFLAGS", "")
```

Comment:
@lizan Is this intentional or just a bug carried over? 
Isn't TARGET_LDFLAGS supposed to be overridden by os.environ.get("LDFLAGS")

was TARGET_LDFLAGS=${CFLAGS} just to address that MacOSX bug mentioned in old code comment?

Reply:
Hmm, not sure, this is just to replicate what we had in `luajit.sh`.

What is the macOS bug, the MACOSX_DEPLOYMENT_TARGET one is not relevant to this environment variable.

Reply:
The
```
-   MACOSX_DEPLOYMENT_TARGET=10.6 DEFAULT_CC=${CC} TARGET_CFLAGS=${CFLAGS} TARGET_LDFLAGS=${CFLAGS} 
```
Was the first place where TARGET_LDFLAGS was overridden with CFLAGS, and I think this small mistake was carried over in
```
+    os.environ["TARGET_LDFLAGS"] = os.environ.get("CFLAGS", "")
```